### PR TITLE
Have `HasTarget` paths be `IsMemberPath`

### PR DESCRIPTION
**Pull request by debs-sifive**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Backend code generation

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by debs-sifive** [comment]
Still testing this locally to make sure this fixes my issue

**Review by mikeurbach — COMMENTED** [review]
I don't think this should be hardcoded one way or another. It should either reflect on the HasTarget to decide if it needs to be a member path or not, or else provide a separate `apply` method like the one that takes a MemBase, which lets the user explicitly indicate if they need a member target.

File: core/src/main/scala/chisel3/properties/Path.scala [hunk 1]
```diff
@@ -94,11 +94,13 @@ object Path {
...
-  def apply(hasTarget: HasTarget): Path = {
+  def apply(hasTarget: HasTarget): Path = apply(hasTarget, false)
+  def apply(hasTarget: HasTarget, isMemberPath: Boolean): Path = {
+    val _isMemberPath = isMemberPath // avoid name shadowing below
     new TargetPath {
       private val scope = Module.currentModule
       def toTarget():   IsMember = hasTarget.toRelativeTarget(scope)
-      def isMemberPath: Boolean = false
+      def isMemberPath: Boolean = _isMemberPath
     }
```
Comment (jackkoenig) [inline core/src/main/scala/chisel3/properties/Path.scala:104]: implementation looks good, can you add a test?

**Comment by debs-sifive** [comment]
urgh idk what i did with the merge stuff

**Comment by debs-sifive** [comment]
ok i think i fixed the rebase 😌

**Review by jackkoenig — APPROVED** [review]

**Pull request merged**